stdlib/http/src/main/ballerina/src/http/auth/authz_filter.bal

```diff
@@ -67,35 +67,28 @@ function handleAuthzRequest(AuthzHandler authzHandler, Request request, FilterCo
     boolean|AuthorizationError authorized = true;
     runtime:Principal? principal = runtime:getInvocationContext()?.principal;
     if (scopes is string[]) {
-        if (scopes.length() > 0) {
-            var canProcessResponse = authzHandler.canProcess(request);
-            if (canProcessResponse is boolean && canProcessResponse) {
-                if(principal is runtime:Principal) {
-                    authorized = authzHandler.process(principal.username, context.getServiceName(),
-                                            context.getResourceName(), request.method, scopes);
-                } else {
-                    authorized = false;
-                }
+        var canProcessResponse = authzHandler.canProcess(request);
+        if (canProcessResponse is boolean && canProcessResponse) {
+            if(principal is runtime:Principal) {
```

Comment:
Don't we need to check this first before the goes to current line 70?

Reply:
Yeah. That reduces the cost in runtime. Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/17439/commits/8ead61f7518af9ece54ba8d0f6ff4cf90b6d410c